### PR TITLE
DAWN-512 - Improve eosioc error message

### PR DESCRIPTION
**Pull request by andriantolie**
Try to improve eosioc error message to be more understandable
screen shot 2018-01-24 at 3 47 49 pm

DAWN-512

### CODE REVIEW

**Comment by andriantolie** [comment]
Merge conflict resolved

**Comment by andriantolie** [comment]
Commits rebased to be more granular and easier for cherry pick

File: programs/eosioc/help_text.cpp [hunk 1]
```diff
@@ -96,9 +96,122 @@ auto smatch_to_variant(const std::smatch& smatch) {
...
+                   "  ],\n"
+                   "  \"actions\" : [{ \"name\":\"foo\",\"type\":\"foo\"}],\n"
+                   "  \"tables\" : [{\n"
+                   "    \"name\":\"foobar_table\",\n"
+                   "    \"index_type\":\"i64\",\n"
+                   "    \"key_names\":[ \"by\" ],\n"
+                   "    \"key_types\":[ \"account_name\" ],\n"
+                   "    \"type\":\"foobar\" "
+                   "  }]\n"
+                   "}"}
```
Comment (heifner) [inline programs/eosioc/help_text.cpp]: These would be much easier to read and maintain if you used: 
```
R"=====(
{
  ...
})=====";
```

Comment (andriantolie) [reply programs/eosioc/help_text.cpp]: Good point. Thanks for the advice! PR Updated

**Comment by andriantolie** [comment]
Merge conflict resolved

**Pull request merged**